src/javascript/esutil.ts

```diff
@@ -284,6 +284,33 @@ export function getReturnFromAnnotation(jsdocAnn: jsdoc.Annotation):
   return type;
 }
 
+/**
+ * Examine the body of a function to see if we can infer something about its
+ * return type. This currently only handles the case where a function definitely
+ * returns void.
+ */
+export function inferReturnFromBody(node: babel.Function): {type: string}|
+    undefined {
+  if (node.async === true || node.generator === true) {
```

Comment:
need to early exit if node is an ArrowFunctionExpression whose body isn't a BlockStatement:

https://astexplorer.net/#/gist/5b1da2824ee4b9b919fb3f7ae4e1e628/2778b732d6cd4ceaf2f4d3ae141a9814c3d8e2f5

Reply:
Done.